tempodb/search/backend_search_block_writer.go

```diff
@@ -0,0 +1,101 @@
+package search
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/grafana/tempo/pkg/tempofb"
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/grafana/tempo/tempodb/encoding"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+// backendSearchBlockWriter is a DataWriter for search data. Instead of receiving bytes slices, it
+// receives search data objects and combintes them into a single FlatBuffer Builder and
+// flushes periodically, one page per flush.
+type backendSearchBlockWriter struct {
+	// input
+	blockID  uuid.UUID
+	tenantID string
+	w        backend.RawWriter
+
+	// vars
+	builder  *tempofb.BatchSearchDataBuilder
+	pageBuf  []byte
+	tracker  backend.AppendTracker
+	finalBuf *bytes.Buffer
+	dw       common.DataWriter
+}
+
+var _ common.DataWriterGeneric = (*backendSearchBlockWriter)(nil)
+
+func newBackendSearchBlockWriter(blockID uuid.UUID, tenantID string, w backend.RawWriter, enc backend.Encoding) (*backendSearchBlockWriter, error) {
+	finalBuf := &bytes.Buffer{}
+
+	dw, err := encoding.LatestEncoding().NewDataWriter(finalBuf, enc)
```

Comment:
I feel like this is "risky" because the latest encoding is a moving target, should we pin this to v2? Similarly for the index writers